### PR TITLE
Add event identifier

### PR DESCRIPTION
**Pull request by christianblais**
@tjoyal this is an attempt at sending a source/identifier down the hole. Name is identifier, although it isn't meant to be unique, i.e. Hooklys can send us the same webhook multiple times. Thoughts?

### CODE REVIEW

File: app/models/rule_event.rb [hunk 1]
```diff
@@ -1,15 +1,17 @@
 class RuleEvent
   def self.load(attributes)
     object = allocate
+    object.instance_variable_set(:@identifier, attributes.fetch("identifier", ""))
```
Comment (christianblais) [inline app/models/rule_event.rb:4]: This line ensures we do not bust on existing events that do not have an identifier yet.

File: app/controllers/callback_controller.rb [hunk 2]
```diff
@@ -1,24 +1,12 @@
...
-    head :ok
-  end
-
   def receive
     CallbackWebhookJob.perform_later(
       shop_domain: request.headers.fetch('X-Shopify-Shop-Domain'),
       callback: params.fetch(:callback).to_unsafe_h,
-      topic: request.headers.fetch('X-Shopify-Topic')
+      topic: request.headers.fetch('X-Shopify-Topic'),
+      shopify_identifier: request.headers.fetch('X-Shopify-Webhook-Id')
```
Comment (tjoyal) [inline app/controllers/callback_controller.rb:9]: I am worried to see the code here grow and grow over time.

Next thing to add will be `X-Hooklys-Id` header and so on.

We could add all headers `headers: headers`? 

One one hand it will mean no longer have to change the callback controller code, on the other it is a waste of memory (which I do not think we care at this point).

Thoughts? I'm good if we want to keep this as-is for now.

```suggestion
      headers: request.headers.to_unsafe_h
```

File: app/models/rule_event.rb (hunk 1 above)
Comment (tjoyal) [reply app/models/rule_event.rb:4]:
```suggestion
    # Ensures we do not bust on historic events that do not have an identifier
    object.instance_variable_set(:@identifier, attributes.fetch("identifier", ""))
```

File: config/routes.rb [hunk 3]
```diff
@@ -9,10 +9,6 @@
     end
   end
 
-  # OLD
-  post '/webhooks/rules/:shop_id/:topic', to: 'callback#webhook', constraints: { shop_id: /\d+/, topic: /.*/ }
```
Comment (tjoyal) [inline config/routes.rb:13]: Let's move this to another PR. We need to make sure no active DB entry have this old entry point configured

**Comment by tjoyal** [comment]
> Name is identifier, although it isn't meant to be unique, i.e. Hooklys can send us the same webhook multiple times. Thoughts?

I would just embrace this fact. Webhooks are delivered "at least once" by design.

File: app/controllers/callback_controller.rb (hunk 2 above)
Comment (christianblais) [reply app/controllers/callback_controller.rb:9]: If we need hooklys id, let's add hooklys id? I'd still prefer to be explicit as opposed to send all of them, although 90% are waste?
